src/Stiefel.jl

```diff
@@ -0,0 +1,239 @@
+using LinearAlgebra: diag, qr, tr, svd, mul!, zeros, lyap
+
+@doc doc"""
+    Stiefel{m,n,T} <: Manifold
+
+The Stiefel manifold consists of all $m\times n$, $m\geq n$ orthonormal matrices, i.e.
+
+````math
+\mathcal M = \{ x \in \mathbb K^{m\times n} : \overline{x}^{\mathrm{T}}x = I_n \}
+````
+where $\mathbb K \in \{\mathbb R, \mathbb C\}$ and
+$I_n \in \mathbb R^{n\times n}$ denotes the unit matrix.
+
+The tangent space at a point $x\in\mathcal M$ is given by
+````math
+T_x\mathcal M = \{ v \in \mathbb{K}^{m\times n} : x^{\mathrm{T}}v + v^{\mathrm{T}}x=0_n\},
+````
+where $0_n$ is the $n\times n$ zero matrix.
+
+The metric is either inherited from $\mathbb R^{m,n}$ for the real-valued case
+or the one inherited from interpreting the complex valued entries in the Gaussian
+plane $\mathbb R^2$ and then ober all entries as before.
```

Comment:
ober?

Reply:
Are the imaginary entries interleaved with the reals or separated?
We should probably set as a convention that equipping a complex manifold with a `EuclideanMetric` means that Hermitian inner product is applied with the Hermitian metric as the identity matrix. Which is equivalent to this definition. If that's documented there, then it doesn't need to be here.

Reply:
Oh, the ober is is a typo (though in German it means “upper” as a prefix, it's just b and v being close on the keyboard).

It is actually the Euclidean (Hermitian) Metric form the embedding. I will rephase that. The question is, whether we should introduce a `HermitianMetric` to emphasise that (the difference could be, that the Euclidean really only does transposed and the Hermitian does an adjoint? Not sure whether that's reasonable).

Reply:
TBH, I don't know enough about complex manifolds to have an informed opinion here. All my use cases use the same Euclidean Hermitian metric. We currently already interpret a `EuclideanMetric` in several ways, all unified by the fact that they have the usual Euclidean inner product. Are there any complex manifolds whose inner product is not Hermitian? If not, maybe the convention should be that all inner products are adjoints, which of course includes the usual transpose.

Reply:
Of course there are, but then a `HermitianMetric` is just not the default metric ;)

My question was more related to the naming scheme, since `EuclideanMetric` might lead a user to think that it just does transpose (not adjoint) on complex numbers (which it of course does not).

Reply:
I think if it's clearly documented, there's not much risk of confusion. And it might be more in line with expectations. e.g. `dot`, which is basically just an alias for "Euclidean inner product", on complex matrices does an adjoint.

Reply:
Ok. Then we just stay with `EuclideanMetric` and I will document that `EuclideanMetric` with a complex manifold means a Hermitian metric.